lib/http2/hpack.c

```diff
@@ -109,7 +116,11 @@ static int contains_invalid_field_name_char(const char *s, size_t len)
     for (; len != 0; ++s, --len) {
         unsigned char ch = (unsigned char)*s;
         if (!valid_h2_field_name_char[ch]) {
-            return 1;
+            if (ch - 'A' < 26U) {
+                *err_desc = err_found_upper_case_in_header_name;
+                return 1;
+            }
+            *err_desc = soft_err_found_invalid_char_in_header_name;
```

Comment:
I think you would also need to adjust the return code as well.

Personally, I do not mind if this path was slow and/or written in two path, since it's an error path, and since it is not the _slowest_ path (I anticipate that HPACK decoder is slower). In other words, this path doesn't get frequently executed, nor is it an attractive target of a DoS attack.